### PR TITLE
detect: Add byte_math detector

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4823

This PR adds `byte_math` to the suite of `byte` functions with a strict implementation based on the Snort documents.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [2311](https://redmine.openinfosecfoundation.org/issues/2311)

Describe changes:
- Address review comments

### CODE REVIEW

File: src/detect-bytemath.c [hunk 1]
```diff
@@ -0,0 +1,1210 @@
...
+
+static int DetectByteMathParseTest02(void)
+{
+    /* bytes value invalid */
+    DetectByteMathData *bmd = DetectByteMathParse(NULL, "bytes 257, offset 2, oper +, "
+                                                  "rvalue 39, result bar", NULL);
+
+    FAIL_IF(bmd != NULL);
+
+    DetectByteMathFree(NULL, bmd);
```
Comment (victorjulien) [inline src/detect-bytemath.c:782]: If read this correctly, this can only get called if `bmd == NULL`?

File: src/detect-bytemath.c [hunk 2]
```diff
@@ -0,0 +1,1210 @@
...
+    PASS;
+}
+
+static int DetectByteMathParseTest02(void)
+{
+    /* bytes value invalid */
+    DetectByteMathData *bmd = DetectByteMathParse(NULL, "bytes 257, offset 2, oper +, "
+                                                  "rvalue 39, result bar", NULL);
+
+    FAIL_IF(bmd != NULL);
```
Comment (victorjulien) [inline src/detect-bytemath.c:780]: `FAIL_IF_NOT_NULL(bmd)` has the same effect, but makes it even more clear that we expect the result to NULL.

File: src/detect-bytemath.c (hunk 1 above)
Comment (victorjulien) [reply src/detect-bytemath.c:782]: If so, this call can be removed.

**Comment by jlucovsky** [comment]
Continued in #4913

**Pull request closed without merging**